src/main/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandler.java

```diff
@@ -196,20 +196,32 @@ public PaymentHandleResponse executePayment(@Nonnull String paypalPlusPaymentId,
                     } else if (PaypalPlusPaymentStates.CREATED.equals(paypalPayment.getState())) {
                         return createChargeTransaction(paypalPayment, ctpPayment.getId(), TransactionState.PENDING);
                     } else {
-                        throw new PaypalPlusPaymentException(format("Error when approving payment [%s], current state=[%s]",
+                        throw new PaypalPlusException(format("Error when approving payment [%s], current state=[%s]",
                                 paypalPlusPaymentId, paypalPayment.getState()));
                     }
                 })
-                .thenApply(payment -> PaymentHandleResponse.ofStatusCode(HttpStatus.OK));
+                .thenApply(payment -> PaymentHandleResponse.ofHttpStatus(HttpStatus.CREATED));
     }
 
-    private PaymentHandleResponse executeWithExceptionallyHandling(String paypalPlusPaymentId,
+    private PaymentHandleResponse executeWithExceptionallyHandling(String paymentId,
+                                                                   String paymentIdType,
                                                                    CompletionStage<PaymentHandleResponse> completionStage) {
         return completionStage
                 .exceptionally(throwable -> {
-                    logger.error("Unexpected exception processing Paypal Plus paymentId=[{}]:", paypalPlusPaymentId, throwable);
+                    logger.error("Unexpected exception processing " + paymentIdType + "=[{}]:", paymentId, throwable);
+                    if (throwable instanceof CompletionException) {
+                        // the real exception is wrapped inside
+                        throwable = throwable.getCause();
+                    }
+                    if (throwable instanceof PaypalPlusServiceException) {
+                        PayPalRESTException restException = ((PaypalPlusServiceException) throwable).getCause();
+                        if (restException != null) {
+                            return PaymentHandleResponse.ofHttpStatusAndErrorMessage(HttpStatus.valueOf(restException.getResponsecode()),
```

Comment:
it might break our defined contract regarding status codes we return if PayPal uses other codes?